### ISSUE TITLE
Add planning stage in mkl_fft 

### ISSUE THREAD

**Issue by sanjiyan3** [IntelPython/mkl_fft#86]
Any plan to add a planning stage ? That is very useful when mkl_fft is  used inside a loop to reduce runtime.

**Comment by oleksandr-pavlyk** [comment]
Yes, actually. In MKL's DFTI interface such plans are called descriptors. `mkl_fft` already caches descriptors for 1D transforms, but it is a good change to expose the committed descriptor as a Python object and allow to pass it to respective Python functions.
